### PR TITLE
Advanced query operator test

### PR DESCRIPTION
**Pull request by mlnewman**
Added unit test coverage for opinion search operators

### CODE REVIEW

File: cl/search/tests.py [hunk 1]
```diff
@@ -334,6 +334,75 @@ def test_issue_729_url_coalescing(self) -> None:
...
+        r = self.client.get(reverse("show_results"), {"q": "Howard !Honda"})
+        self.assertIn("had no results", r.content.decode())
+
+        r = self.client.get(reverse("show_results"), {"q": "Howard -Honda"})
+        self.assertIn("had no results", r.content.decode())
+
+    def test_query_phrase(self) -> None:
+        """Can we query by phrase"""
+        r = self.client.get(
+            reverse("show_results"), {"q": "Testing Supreme Court"}
```
Comment (mlissner) [inline cl/search/tests.py:381]: This doesn't quite look like a phrase search to me. Shouldn't it have encoded quotes or something like that?

**Review by mlissner — COMMENTED** [review]
Looks good. Nice and simple. One little question. Are you planning to add additional advanced tests in here as well to test the other search features?

File: cl/search/tests.py (hunk 1 above)
Comment (mlnewman) [reply cl/search/tests.py:381]: Sure I can add the others. Wild card, date and range tests already exist. I thought there was a proximity ('"net neutrality"~2', '"net neutrality"~2') test but I can add something more specific. I believe the only missing tests are related and fielded.

That said, I will just update AdvancedTest with all of the operations represented here to make things more concise.

https://www.courtlistener.com/help/search-operators/

Comment (mlissner) [reply cl/search/tests.py:381]: That's great, thanks Marion. When it's ready, I think you can use the "Request Review" feature again or flag me to look again and I'll take another look.

**Comment by mlnewman** [comment]
@mlissner Please have another look at this PR. Didn't see option to request another review.

File: cl/search/tests.py [hunk 2]
```diff
@@ -334,6 +334,127 @@ def test_issue_729_url_coalescing(self) -> None:
...
+        r = self.client.get(reverse("show_results"), {"q": "Howard !Honda"})
+        self.assertIn("had no results", r.content.decode())
+
+        r = self.client.get(reverse("show_results"), {"q": "Howard -Honda"})
+        self.assertIn("had no results", r.content.decode())
+
+    def test_query_phrase(self) -> None:
+        """Can we query by phrase"""
+        r = self.client.get(
+            reverse("show_results"), {"q": "'Testing Supreme Court'"}
```
Comment (mlissner) [inline cl/search/tests.py:381]: Could you prove the reverse here too? For example `{"q": "'Supreme Testing Court'"}` should bring back zero results, right?

**Review by mlissner — CHANGES_REQUESTED** [review]
This looks great. One small comment, if you don't mind. I'm just raising it because I worry that the test isn't quite testing what we think it is and the additional test might flush out a mistake (or maybe not, we'll know soon!).

**Comment by mlissner** [comment]
Thank you @mlnewman. Congrats on your first merge!

**Pull request merged**